### DISCUSSION TITLE
How to get environment variable in Windows batch-file

### DISCUSSION THREAD

**Discussion by AxelPetermann** [watchexec/watchexec#879] [Questions & Help] [answered]
Hello,

maybe this is a stupid beginners-question, but I would like to know how to read the environment variable $WATCHEXEC_EVENTS_FILE when the command of watchexec is a Windows batch-file.

My code currently looks like the following (in a bat-file):
```
watchexec.exe --postpone --emit-events-to=file --watch %watchFolder% "D:\command.bat" "$WATCHEXEC_EVENTS_FILE"
```
And the `command.bat` is (just would like to see the path where $WATCHEXEC_EVENTS_FILE is stored):
```
echo %*
```
But the output is always (German Windows):
```
ECHO ist ausgeschaltet (OFF).
```

**Comment by passcod** [comment]
Several things here. First you should enable env vars if you want them, or the json features instead. Default is not to pollute environment.

Second `echo whatever` is a unix shell idiom. In Windows Batch `echo` toggles the printout mode, which prints "echo is on" or "echo is off" in your locale. So you'll need to figure out the correct thing here.

**Reply by AxelPetermann** [reply]
> First you should enable env vars if you want them

Interesting. Haven't seen that I need to enable env vars. Looked into the help-page, but coundn't figure out how to enable them...

**Reply by passcod** [reply]
--emit-events-to

**Comment by AxelPetermann** [comment]
Hi passcod,

thanks again for your quick answer.

In my first post I've used "--emit-events-to=file", but still couldn't figure out how to get the content of the env variable.

BTW: In Windows bat-files the "echo" command can be used to display a message or to display the value of a variable ([echo-command | geeksforgeeks](https://www.geeksforgeeks.org/batch-script-echo-command/))

**Reply by passcod** [reply]
Okay. So, you see, when you set `--emit-events-to=file`, it emits events to a file. Not to the environment.

**Comment by AxelPetermann** [comment]
Maybe I was not that precise with my question.

I know that `--emit-events-to=file` emits events to a file. And the path of the file is stored in the env variable `$WATCHEXEC_EVENTS_FILE`. Now, I want to use a `*.bat` file as `[COMMAND]`.

My question now is: How can I further process (or as a first step read) the content of the env variable `$WATCHEXEC_EVENTS_FILE` within that `*.bat` file (I don't know how to read the env variable in the bat-file).

**Reply by passcod** [reply, marked as answer]
```bat
type %WATCHEXEC_EVENTS_FILE%
```

**Reply by AxelPetermann** [reply]
Thanks, that did it.
